src/Sulu/Bundle/ContactBundle/Controller/AccountController.php

```diff
@@ -391,6 +391,7 @@ public function cgetAction(Request $request)
         if ('true' == $request->get('flat')) {
             $fieldDescriptors = $this->getFieldDescriptors();
             $listBuilder = $this->generateFlatListBuilder();
+            $listBuilder->distinct(true);
```

Comment:
without enable `DISTINCT`, accounts with multiple tags will be duplicated in the response if the tag filter is active

Reply:
TODO: add a test cases for the controllers to check this

Reply:
this is necessary because all joins of all expressions fields are added to the query by the `DoctrineListBuilder` at the moment:
https://github.com/sulu/sulu/blob/7833cdced3878fe25303c0f93c99daeba1e5e6b8/src/Sulu/Component/Rest/ListBuilder/Doctrine/DoctrineListBuilder.php#L370

i am not sure if there is a reason for this yet. as far as i understand, it should be enough to add the joins for `searchFields` and `expressionFields` in the query that selects the ids only:
https://github.com/sulu/sulu/blob/2.x/src/Sulu/Component/Rest/ListBuilder/Doctrine/DoctrineListBuilder.php#L267-L300